### DISCUSSION TITLE
"Export changes to a branch" Fails After Codespace Credit Exhaustion

### DISCUSSION THREAD

**Discussion by AliAryanTech** [community/community#147390] [Codespaces] [answered]
### Select Topic Area

Bug

### Body

Hi GitHub Community,  

I recently encountered an issue with Codespaces and would like to report it as a potential bug.  

This month, I used up all my Codespace credits, and unfortunately, a project I was working on was shut down as a result. I noticed an option labeled **"Export changes to a branch"**, which seemed like a great way to save my progress. However, when I tried to use this feature, it took an unusually long time to process, and after about 10 minutes, it failed with no clear explanation or resolution.  

Below are screenshots that illustrate the issue:  
![Screenshot 2024-12-18 164518](https://github.com/user-attachments/assets/d04a23f2-d0b1-4f8a-b37e-ee9e4c0206d8)
...
still on loading...
<img width="592" alt="Screenshot 2024-12-18 164613" src="https://github.com/user-attachments/assets/40d1f3b2-ecf4-412e-a132-dfa566078dbb" />

This behavior is frustrating as it risks losing unsaved progress. If this is a bug, I kindly request that it be investigated and resolved.  
Tagging @Cbodfield for visibility since I understand they work at GitHub and might be able to assist or direct this to the relevant team.  

Thanks in advance for your help!

**Comment by Cbodfield** [comment]
Hey @AliAryanTech! 

I'm sorry to hear you've run into some problems exporting to a branch but thank you for letting us know. I've passed this along to our Codespaces team to investigate.

**Reply by Rajveer812** [reply]
i am also facing same issue i am not able to export .firstly its taking more than 10 minutes then says export do not complete please try again later please help me

**Comment by dmgardiner25** [comment]
Hi @AliAryanTech, thanks for the report and apologies for the inconvenience. We've tracked things down and have identified the issue and will reach back when a fix is available!

**Reply by AliAryanTech** [reply]
Hi, thanks for the update!
But it's been over a day and the issue is still unresolved, The delay is affecting my project so prioritize this asap.

**Reply by MirilRetry** [reply]
It's been 2 weeks and I'm still having the same problem, any updates? I need to export my changes

**Reply by AliAryanTech** [reply]
idk what they doing, but got new credits so just pushed my changes

**Reply by dmgardiner25** [reply, marked as answer]
Exporting should now work. Please give it a try and let me know if you are still running into issues with it.

**Reply by MirilRetry** [reply]
> Exporting should now work. Please give it a try and let me know if you are still running into issues with it.

It works now, Thanks!

**Comment by YaroslavSamchuk** [comment]
I have the same thing, there is a lot of my new code in my codespace. i wait 1 day and i have the same problem.
![ImageGithubIssue](https://github.com/user-attachments/assets/8c18893b-3854-4995-8409-7e8be01eb816)

**Comment by LorenRd** [comment]
I'm also interested on this fix, It's happening the same to me.

Thanks!

**Comment by Aidnpatrick** [comment]
Same here, it won't let me create new branch. I'm very worried for my code.😟 If anyone has a solution, I would like to know.

**Comment by aymnalzahrani7** [comment]
im facing this problem a month ago !!

![Screenshot 2024-12-25 171355](https://github.com/user-attachments/assets/e55cbc38-8e15-4181-93ec-1f0f4000d4c8)
![Screenshot 2024-12-25 172035](https://github.com/user-attachments/assets/073862a1-820a-454f-9534-9bee37480cb0)

@AliAryanTech 
@Cbodfield
